lib/ProductOpener/Ingredients.pm

```diff
@@ -3368,23 +3368,47 @@ sub get_missing_ecobalyse_ids ($ingredients_ref) {
 		delete $ingredient_ref->{ecobalyse_code};
 		delete $ingredient_ref->{ecobalyse_proxy_code};
 
-		# Getting properties from product
-		#my $product_origin = has_specific_ingredient_property()
-
-		# Getting the correct ecobalyse comake devde
-		my $ecobalyse_code = get_inherited_property("ingredients", $ingredient_ref->{id}, "ecobalyse:en");
-		if (defined $ecobalyse_code) {
-			$ingredient_ref->{ecobalyse_code} = $ecobalyse_code;
-		}
-		else {
-			my $ecobalyse_proxy_code
-				= get_inherited_property("ingredients", $ingredient_ref->{id}, "ecobalyse_proxy:en");
-			if (defined $ecobalyse_proxy_code) {
-				$ingredient_ref->{ecobalyse_proxy_code} = $ecobalyse_proxy_code;
-			}
-			else {
-				push(@ingredients_without_ecobalyse_ids, $ingredient_ref->{id});
+		# We are now looking for the appropriate ecobalyse id :
+		# ecobalyse_origins_france_label_organic (if the product comes from france, and is organic)
+		# ecobalyse_origins_european-union_label_organic (if the product comes from europe, and is organic)
+		# ecobalyse_label_organic (if the product is organic)
+		# ecobalyse_origins_france (if the product comes from france)
+		# ecobalyse_origins_european-union (if the product comes from the Europe region)
+		# ecobalyse (else)
+
+		# List of suffixes
+		my @suffixes = ();
+		# If the ingredient is organic...
+		if ($ingredient_ref->{labels} =~ /\ben:organic\b/) {
+			push @suffixes, "_labels_en_organic";
+		}
+		push @suffixes, '';
+
+		if ($ingredient_ref->{labels} =~ /\ben:organic\b/) {
+			push @suffixes, "_proxy_labels_en_organic";
```

Comment:
Instead of adding _proxy_* suffixes that are the same as the non proxy suffixes, it's probably better to have only the non proxy suffixes:
- we first loop on it to see if we have an exact match, and if so we set ecobalyse_code
- if we don't have a match, we loop on it again (and add the proxy_), if we set ecobalyse_proxy_code